### DISCUSSION TITLE
Issues in finding effective diffusivity and tortuosity using OpenPNM

### DISCUSSION THREAD

**Discussion by utopiankk** [PMEAL/OpenPNM#2900] [Q&A]
Dear Professor,
When I used Openpnm to calculate the effective diffusivity and tortuosity of an imported porous medium, the effective diffusivity, D_eff, was smaller than the original air diffusivity, D_AB, which resulted in  the tortuosity calculation that is less than 1. And the diffusivity of the porous medium should be lower than that outside the porous medium. I suspect this is due to an issue in the calculation of effective diffusivity, possibly arising from the calculation "rate_inlet = fd.rate(pores=inlet)[0]". Is there any solution to this issue? Here is my code.
[D_eff.txt](https://github.com/PMEAL/OpenPNM/files/14525721/D_eff.txt)

**Comment by jgostick** [comment]
This usually happens when you have problems with the geometry of the system, like pores that overlap too much and result in negative diffusive conductance.  You can try using the cones_and_cylinders geometry models which handles overlapping pores better.  You can also check to see if any throat lengths are negative.

**Reply by utopiankk** [reply]
Thank you for your quick response, Professor. I have switched the geometric model to cones_and_cylinders, and it does work. However, since I reconstructed the gas diffusion layer model using PoreSpy, the porosity is relatively larger compared to porous structures like rocks, resulting in some overlapping pores. To address this, I've used the inscribed diameter as the diameter for both pores and throats to avoid pore overlap. This seems to cause a significant reduction in the porosity of the pore network model compared to the actual porosity. Does OpenPNM have any good solutions for this issue?

**Reply by jgostick** [reply]
High porosity and anisotropic materials are a challenge, and GDLs are both.  You can do it, but you must be careful.  PNMs are not great for matching porosity for reasons you mention, so I would not use that as a metric to validate your network.  If you're doing transport calculations the pore volume is not relevant, only the conductance is.  Conductance models work because they don't care about overlap, they only care about the transport from A and B.
